### DISCUSSION TITLE
Is it considered an anti-pattern access one store in another directly

### DISCUSSION THREAD

**Discussion by rzec-allma** [pmndrs/zustand#2310] [Q&A]
So something I am finding myself doing is accessing one store in another like this:
```ts
const notificationId = useGlobalNotifications.getState().addNotification(/* ... */);
```
While I find this pattern incredibility useful in reducing boilerplate style code and reducing the amount of code in the component itself, it feel anti-react for some reason (I feel like the react way would be to have this code in the component as a `useCallback()` or something for some reason).

Is this pattern concerned an anti-pattern? Are there any issues or gotcha's that I might run into with this pattern?

**Comment by dbritto-dev** [comment]
@rzec-allma `getState` works well when we talk about sync operations but you could expect some weird behaviors if we talk about async operations you would ended up dealing with old states, if you need to access one store in another then you need merge those stores into one and use slice pattern

**Comment by dai-shi** [comment]
I think it's fine. See also: https://github.com/pmndrs/zustand/blob/27bffb1fba9de4c958f4a4b4d31c20e54e579fc0/docs/guides/practice-with-no-store-actions.md
